insights/client/connection.py

```diff
@@ -1127,6 +1127,10 @@ def checkin(self):
         '''
         logger.info("Checking in...")
 
+        if not self.config.legacy_upload and not self.api_registration_check():
```

Comment:
I see that using `--checking` switches automatically to non-legacy upload; regardless, I'd remove the explicit check here to reduce the checks for legacy upload.
Rather than using `api_registration_check()` directly, use `insights.client.support.registration_check()`, as it should take care of whatever upload mode (legacy vs non-legacy).